src/main/kotlin/org/rust/lang/core/psi/ext/RsAbstractable.kt

```diff
@@ -22,6 +22,18 @@ sealed class RsAbstractableOwner {
     val isInherentImpl: Boolean get() = this is Impl && isInherent
     val isTraitImpl: Boolean get() = this is Impl && !isInherent
     val isImplOrTrait: Boolean get() = this is Impl || this is Trait
+
+    val crateRelativePath: String?
+        get() = when (this) {
+            is Trait -> RsPsiImplUtil.crateRelativePath(trait)
+            is Impl -> {
+                when (val typeReference = impl.typeReference) {
+                    is RsBaseType -> typeReference.path?.referenceName.let { "${impl.containingMod.crateRelativePath}::$it" }
```

Comment:
Also, the current implementation is confusing
Imagine the following case:
```rust
//- main.rs
mod bar;

struct Foo;
trait Baz {
    fn baz(&self);
}
trait FooBar {
    fn foo_bar(&self);
}

fn main() {
    Foo.foo(); // `foo` -> `bar::Foo::foo`
    123.foo_bar(); // `foo_bar` -> `bar::T::foo_bar`
    (&123).baz(); // `baz` -> `bar::baz`
}

//- bar.rs
use crate::{Foo, Baz, FooBar};
impl Foo {
    pub fn foo(&self) {}
}

impl Baz for &i32 {
    fn baz(&self) {}
}

impl<T> FooBar for T {
    fn foo_bar(&self) {}
}
```

The current implementation gives `bar::Foo::foo` for `foo` method of `Foo` struct, i.e. adds struct name to crate relative path of a module with `impl` - it looks ok in general, although I don't know if it's useful.
For `foo_bar` it provides `bar::T::foo_bar` and it's weird. What's `T`? Of course, as author of code, I know that it's `impl` type parameter but how can I distinguish this case from previous where `Foo` is a struct?
In the case of `baz` method for `&i32`, we omit type name at all. I think it can be very confusing for a user.